## Importing Libraries

In [ ]:
import pandas as pd              #Useful for loading the data set
import matplotlib.pyplot as plt

## Load Data

In [2]:
dataset=pd.read_csv("data.csv")

In [3]:
dataset

hours  age  internet  marks
0     6.83   15         1  78.50
1     6.56   16         0  76.74
2      NaN   17         1  78.68
3     5.67   18         0  71.82
4     8.67   19         1  84.19
..     ...  ...       ...    ...
196   8.56   19         1  84.68
197   8.94   20         1  86.75
198   6.60   15         1  78.05
199   8.35   16         1  83.50
200   4.15   15         0  81.45

[201 rows x 4 columns]

- dataset variable set hold the complete information of dataset.

In [4]:
print(dataset.shape)
print(dataset.head(5))

(201, 4)
   hours  age  internet  marks
0   6.83   15         1  78.50
1   6.56   16         0  76.74
2    NaN   17         1  78.68
3   5.67   18         0  71.82
4   8.67   19         1  84.19


- Here we get the complete information of dataset 
- dataset(dataset.shape) - It shows 201 Columns and 4 Rows
- dataset(dataset.shape) - It shows top 5 results of the dataset

## Finding and removing NA values from our Features X

In [6]:
dataset.columns[dataset.isna().any()]

Index(['hours'], dtype='object')

In [8]:
dataset.hours=dataset.hours.fillna(dataset.hours.mean())

In [9]:
dataset.hours

0      6.830000
1      6.560000
2      6.981429
3      5.670000
4      8.670000
         ...   
196    8.560000
197    8.940000
198    6.600000
199    8.350000
200    4.150000
Name: hours, Length: 201, dtype: float64

- Here we check null values from our Features X and fill it.

## Segregate data into X and Y

In [12]:
X=dataset.iloc[:,:-1].values

In [13]:
X

array([[ 6.83      , 15.        ,  1.        ],
       [ 6.56      , 16.        ,  0.        ],
       [ 6.98142857, 17.        ,  1.        ],
       [ 5.67      , 18.        ,  0.        ],
       [ 8.67      , 19.        ,  1.        ],
       [ 7.55      , 20.        ,  0.        ],
       [ 6.67      , 15.        ,  0.        ],
       [ 8.99      , 16.        ,  0.        ],
       [ 5.19      , 17.        ,  1.        ],
       [ 6.75      , 18.        ,  0.        ],
       [ 6.59      , 19.        ,  0.        ],
       [ 8.56      , 20.        ,  1.        ],
       [ 7.75      , 15.        ,  0.        ],
       [ 7.9       , 16.        ,  1.        ],
       [ 8.19      , 17.        ,  0.        ],
       [ 6.55      , 18.        ,  1.        ],
       [ 6.36      , 19.        ,  0.        ],
       [ 8.44      , 20.        ,  1.        ],
       [ 8.41      , 15.        ,  0.        ],
       [ 7.67      , 16.        ,  1.        ],
       [ 7.42      , 17.        ,  1.   

- Here we got the values of X

In [16]:
Y=dataset.iloc[:,-1].values

In [17]:
Y

array([78.5 , 76.74, 78.68, 71.82, 84.19, 81.18, 76.99, 85.46, 70.66,
       77.82, 75.37, 83.88, 79.5 , 80.76, 83.08, 76.03, 76.04, 85.11,
       82.5 , 80.58, 82.18, 83.36, 70.67, 75.02, 70.96, 83.33, 74.75,
       75.65, 74.15, 80.17, 82.27, 76.14, 71.1 , 84.35, 83.08, 76.76,
       81.24, 78.21, 73.08, 83.23, 70.27, 86.41, 71.1 , 82.84, 82.38,
       72.96, 77.46, 70.11, 72.38, 71.41, 72.22, 77.77, 84.44, 71.45,
       82.21, 85.48, 75.03, 86.65, 70.9 , 71.7 , 73.61, 79.41, 76.19,
       80.43, 85.78, 70.06, 81.25, 81.7 , 69.27, 82.79, 71.8 , 71.79,
       74.97, 78.61, 77.59, 72.33, 72.08, 77.33, 70.05, 73.34, 84.  ,
       82.93, 76.63, 75.36, 77.29, 72.87, 73.4 , 81.74, 71.85, 84.6 ,
       79.56, 82.1 , 72.08, 79.1 , 81.01, 76.48, 75.39, 68.57, 83.64,
       82.3 , 75.18, 82.03, 82.99, 79.26, 77.55, 77.07, 72.1 , 73.25,
       74.25, 70.58, 81.08, 75.04, 76.38, 80.86, 78.42, 74.44, 70.34,
       85.04, 73.61, 75.55, 76.2 , 82.69, 76.83, 79.53, 83.57, 85.95,
       76.02, 77.65,

In [ ]:
- Here we got the values of Y

## Training Dataset using Linear Regression

In [18]:
from sklearn.linear_model import LinearRegression

In [19]:
model=LinearRegression()

In [20]:
model.fit(X,Y)

LinearRegression()

- Here we are using Linear Regression for our data set

## predicted price for land sq.feet of custom values 

In [24]:
a=[[24,27,5]]
PredictedModelResult=model.predict(a)
print(PredictedModelResult)

[141.58023867]


- Conclusion - predicted price for land sq.feet of custom values is 141.58023867